### PR TITLE
fix generator tool

### PR DESCRIPTION
**Pull request by Alancere**
current version: v3.0.0-beta.1
to release: v2.7.0 (compare v2.6.0, not BreakingChange)

The above does not replace the import module in fake and live_test.
ref: https://github.com/Azure/azure-sdk-for-go/pull/22184/commits/b8368c704ce55b61adfc37894e6786d47b5aedac#diff-878839ecfc6b682d3adbdd34fdeff0178e7018301d85ea4dca7453563d9ca93c

### CODE REVIEW

File: eng/tools/generator/cmd/v2/common/fileProcessor.go [hunk 1]
```diff
@@ -566,11 +565,25 @@ func replaceModuleImport(path, rpName, namespaceName, previousVersion, currentVe
...
 
+func getModuleVersion(autorestPath string) (*semver.Version, error) {
+	data, err := os.ReadFile(autorestPath)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, line := range strings.Split(string(data), "\n") {
+		if strings.HasPrefix(line, autorest_md_module_version_prefix) {
+			return semver.NewVersion(strings.TrimSpace(line[len(autorest_md_module_version_prefix):]))
```
Comment (raych1) [inline eng/tools/generator/cmd/v2/common/fileProcessor.go:580]: Will `NewVersion` return an error when the value is not a valid version?

Comment (Alancere) [reply eng/tools/generator/cmd/v2/common/fileProcessor.go:580]: yes

Comment (raych1) [reply eng/tools/generator/cmd/v2/common/fileProcessor.go:580]: What I meant is if NewVersion returns error on an invalid value then should we customize the error message before returning?

**Review by raych1 — APPROVED** [review]

File: eng/tools/generator/cmd/v2/common/fileProcessor.go (hunk 1 above)
Comment (Alancere) [reply eng/tools/generator/cmd/v2/common/fileProcessor.go:580]: There will be no invalid versions in autorest.md, and the program should be stopped if the module-version in autorest.md is invalid

**Pull request merged**